contractcourt/channel_arbitrator.go

```diff
@@ -1135,11 +1146,141 @@ func (c *ChannelArbitrator) sweepAnchors(anchors *lnwallet.AnchorResolutions,
 		if err != nil {
 			return err
 		}
+
+		return nil
+	}
+
+	// Sweep anchors based on different HTLC sets. Notice the HTLC sets may
+	// differ across commitments, thus their deadline values could vary.
```

Comment:
I get why you do this, but can they really vary? I thought they had the same CLTV on both commitments always.

Reply:
Yeah I think it's very unlikely. Even there's a diff between remote and local commit txes, I think the local and remote should be the same?
But still, because we have three sets of HTLCs and three commit txes here, structurally we should have three deadlines. If they are the same, then good. If not, maybe something went wrong.